### DISCUSSION TITLE
[HELP] 1.12 Beta 6 - Templater Script for Quick Tools

### DISCUSSION THREAD

**Discussion by FelipeRearden** [chrisgurney/obsidian-note-toolbar#126] [Q&A] [answered]
Hello @chrisgurney !!!

I am trying to build a workflow around a templater script that is not working 

Workflow: open the NTB modal with the filled with the number `4`

```
<%*
app.commands.executeCommandById('note-toolbar:open-quick-tools')
await new Promise(r => setTimeout(r,3000));
activeWindow.activeDocument.querySelector(".prompt.note-toolbar-setting-item-suggester-dialog").dispatchEvent(new KeyboardEvent("keydown", { key: "4" }));
%>
```

I am struggling to target the  `.prompt-input` to mimic a keyboard input.

### IDEA

My idea is to open the modal filled by items a certain criteria (in this example all items with the letter `4` )

This way I can create a complex templater script where I open NTB modal filtered by a subset of items that has the same prefix 

📷
![IMG_1294](https://github.com/user-attachments/assets/aee09c8e-13d2-4d66-bf0b-48a16ec2a0b2)

Forgive for the ask, I know this is not 100% to NTB, but could you help me on this one ? 🙏

Thank you very much ☀️

**Comment by chrisgurney** [comment, marked as answer]
@FelipeRearden You were on the right track! The below seems to work:

```js
<%*
app.commands.executeCommandById('note-toolbar:open-quick-tools');
await new Promise(r => setTimeout(r,1000));
var input = activeWindow.activeDocument.querySelector(".note-toolbar-setting-item-suggester-dialog .prompt-input");
input.value = "4";
input.dispatchEvent(new Event("input"));
%>
```

- The selector is the `input` element/field itself
- Set the value of the input field
- Dispatch an "input" event to tell the form there's something there, I guess?
- I also reduced the timeout to 1s (though this may need to vary by device?) above, just so it's quicker

**Reply by FelipeRearden** [reply]
This is absolute perfect @chrisgurney !!!! 🎉 🎉 🎉

Many many many thanks for this snippet. This is part of a workflow for `Quick Tools` that I want to create and share with you in the next few days. 

Thank you very much for the explanation. Dealing with mimic keyboard events is something that I really don’t understand enough to try to do alone.

You snippet will be useful for other purposes as well 🙏

> Reduced the timeout to 1s (though this may need to vary by device?)

Yeah, I changed for 200ms. When testing I was using 2000ms to get time to make sure that it was time enough to input `4` 😀

<hr> 

Thanks one more time Chris ☀️

**Comment by FelipeRearden** [comment]
@chrisgurney let me just ask another question that is related to this workflow.

I am asking here to avoid confusion with `1.12 beta`. If you like the idea, I can open a new comment or issue. 

Based on this idea if opening Quick Tools with `4` as a starter input…

QUESTION: would be possible to have a setting on Settings to make the `search input` NOT FUZZY SEARCH ?

Let me explain with as example :

- I will use `CMD+4` as the keyboard shortcut for Quick Tools (My finger in already placed on the key 4)
- Based on a Templater Script I will use a different number for `input.value` 
- Each NTB will have a different number as a prefix for the label. Example:
  - NTB Book Reading has all items will start with the prefix `1`  -> [1,14,144,1444]
  - NTB Article Reading has all items will start with the prefix `2` -> [2,24,244,2444]

- When I hit `CMD+4` + `input.value = 1` my modal will be filtered  with only `[1]` item 
- To go to the next item I hit `4` since my finger is already there 🙏
- Follow the same rule to next item…
- Hit ENTER to execute the chosen item.

<hr> 

if the Modal is not FUZZY I could get the exact NTB without any fear of getting items from another group of toolbars since `4` is the common number between these NTB labels. 

This is a WIP just for you to understand my vision feature: I want a fast way to cycle between the Quick Tools modal filtered by a criteria in a custom modal that is almost invisible in the workspace.. Something like this
- In my main vault will have a different CSS styke since I dont have box-shadow and `modal-bg`. It is just for you to get the concept

https://github.com/user-attachments/assets/1f859da4-13bc-4419-a1a8-52fc8200e595



💡 That is why I need the NOT FUZZY. This modal would be a single item almost invisible in my workspace changing the item based on the times that I hit the key `4` in the keyboard

<hr>

Would be possible to achieve this ?

**Comment by FelipeRearden** [comment]
@chrisgurney  the solution was I front of my eyes all the time …

By using this CSS snippet, I was able to mimic the NOT FUZZY behavior:

```css
.prompt.note-toolbar-setting-item-suggester-dialog .suggestion-item.note-toolbar-item-suggestion:not(.is-selected) {
	display: none;
}
```

By not hidding only the active suggestion item, I can have a single item all the time 🙏

**Comment by FelipeRearden** [comment]
I am updating the script with the new syntax for the Quick Tools command on `beta-16`

```
<%*
app.commands.executeCommandById('note-toolbar:open-item-suggester')
await new Promise(r => setTimeout(r,100));
var input = activeWindow.activeDocument.querySelector(".note-toolbar-setting-item-suggester-dialog .prompt-input");
input.value = "4";
input.dispatchEvent(new Event("input"));
%>
```
